File: app/src/main/java/com/jeeny/task/ui/map/adapter/VehicleSpinnerAdapter.java
package com.jeeny.task.ui.map.adapter;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import androidx.annotation.NonNull;

import org.jetbrains.annotations.NotNull;

import java.util.List;


public class VehicleSpinnerAdapter extends ArrayAdapter<String> {

    private Context context;
    private int resourceID;

    public VehicleSpinnerAdapter(@NonNull Context context, int resourceID, @NonNull List<String> categories) {
        super(context, resourceID, categories);
        this.context = context;
        this.resourceID = resourceID;
    }


    @NotNull
    @Override
    public View getView(int position, View convertView, @NotNull ViewGroup parent) {
        return rowView(convertView, position);
    }

    @Override
    public View getDropDownView(int position, View convertView, @NotNull ViewGroup parent) {
        return rowDropDownView(convertView, position);
    }

    private View rowView(View convertView, int position) {
        viewHolder holder;
        View rowView = convertView;
/*         if (rowView == null) {
            holder = new viewHolder();
            LayoutInflater inflater = (LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            rowView = inflater.inflate(R.layout.category_spinner_item, null, false);
            holder.txtTitle = (TextView) rowView.findViewById(R.id.itemCat_title);
            rowView.setTag(holder);
        } else
            holder = (viewHolder) rowView.getTag();
        holder.bindViews(categories.get(position));*/
        return rowView;
    }

    private View rowDropDownView(View convertView, int position) {
        viewHolderDropDown holder;
        View rowView = convertView;
     /*   if (rowView == null) {
            holder = new viewHolderDropDown();
            LayoutInflater inflater = (LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            rowView = inflater.inflate(R.layout.category_spinner_drop_down_item, null, false);
            holder.txtTitle = (TextView) rowView.findViewById(R.id.itemCat_dropDown);
            rowView.setTag(holder);
        } else
            holder = (viewHolderDropDown) rowView.getTag();
        holder.bindViews(categories.get(position));*/
        return rowView;
    }

    private class viewHolder {
        TextView txtTitle;

        public void bindViews(String category) {
            // txtTitle.setText(category.getName());
        }
    }

    private class viewHolderDropDown {
        TextView txtTitle;

        public void bindViews(String category) {
            //   txtTitle.setText(category.getName());
        }
    }
}

File: app/src/main/java/com/jeeny/task/ui/map/adapter/VehicleSpinnerAdapter.kt
/*
 * Created by Muhammad Mehroz Afzal on 2020.
 */

package com.jeeny.task.ui.map.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.jeeny.task.R
import com.jeeny.task.repository.model.PoiListItem

class VehicleSpinnerAdapter(
    context: Context,
    resourceID: Int,
    private val types: List<String>
) : ArrayAdapter<String>(
    context, resourceID, types
) {
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        return rowView(convertView, position)
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View? {
        return rowDropDownView(convertView, position)
    }

    private fun rowView(convertView: View?, position: Int): View? {
        val holder: ViewHolder
        var rowView = convertView
        if (rowView == null) {
            holder = ViewHolder()
            val inflater =
                context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            rowView = inflater.inflate(R.layout.item_vehicle, null, false)
            holder.txtTitle = rowView.findViewById<View>(R.id.itemVehicle_type) as TextView
            rowView.tag = holder
        } else holder = rowView.tag as ViewHolder
        types[position].let { holder.bindViews(it) }
        return rowView
    }

    private fun rowDropDownView(convertView: View?, position: Int): View? {
        val holder: ViewHolderDropDown
        var rowView = convertView
        if (rowView == null) {
            holder = ViewHolderDropDown()
            val inflater =
                getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            rowView = inflater.inflate(R.layout.item_vehicle_expanded, null, false)
            holder.txtTitle = rowView.findViewById<View>(R.id.itemVehicleExpanded_type) as TextView
            rowView.tag = holder
        } else holder = rowView.tag as ViewHolderDropDown
        types[position].let { holder.bindViews(it) }
        return rowView
    }

    private inner class ViewHolder {
        var txtTitle: TextView? = null
        fun bindViews(item: String) {
            txtTitle?.text = item
        }
    }

    private inner class ViewHolderDropDown {
        var txtTitle: TextView? = null
        fun bindViews(item: String) {
            txtTitle?.text = item
        }
    }
}